src/update_handler.c

```diff
@@ -132,16 +133,25 @@ static gboolean copy_raw_image(RaucImage *image, GUnixOutputStream *outstream, G
 	g_unix_output_stream_set_close_fd(outstream, FALSE);
 
 	size = g_output_stream_splice((GOutputStream *) outstream, instream,
-			G_OUTPUT_STREAM_SPLICE_CLOSE_SOURCE | G_OUTPUT_STREAM_SPLICE_CLOSE_TARGET,
+			G_OUTPUT_STREAM_SPLICE_CLOSE_TARGET,
 			NULL,
 			&ierror);
 	if (size == -1) {
 		g_propagate_prefixed_error(error, ierror,
 				"Failed splicing data: ");
 		return FALSE;
-	} else if (size != (gssize)image->checksum.size) {
+	}
+
+	seeksize = g_seekable_tell((GSeekable*) instream);
+
+	if (seeksize != (goffset)image->checksum.size) {
```

Comment:
checksum.size is a gsize, which is 32bit long on 32bit architectures. So this would still fail on images >= 4GB. We should use goffset in the struct and everywhere we use gsize for file offsets/sizes as well.

Reply:
The difference of `gsize` and `goffset` is that `goffset` is signed which is useless for values that should not become negative. Thus maybe we should move to `guint64` instead? This is the same that the mbr switching implementation already uses.

Reply:
We've just discussed this internally, and will go with `goffset` for file sizes in general.